ast2vec/repo2/base.py

```diff
@@ -126,22 +127,41 @@ def thread_loop(thread_index):
                             break
                         try:
                             folder, filename, language = task
-                            filepath = os.path.join(folder, filename).encode('utf8')
+                            filepath_naive = os.path.join(folder, filename).encode('utf8')
                             # I need .encode('utf8') to avoid problems
                             # with bad symbols in file names on Ubuntu
 
                             # Check if file path is symlink
-                            if os.path.islink(filepath):
-                                filepath = os.readlink(filepath)
+                            islink = os.path.islink(filepath_naive)
+                            if islink:
+                                filepath = os.readlink(filepath_naive)
+                            else:
+                                filepath = filepath_naive
+                            if not os.path.exists(filepath):
+                                self._log.warning('File %s does not exist.\n\t%s',
```

Comment:
yes, I do not like it too.
I think to put it out from such a deep nesting, but not sure it is a good idea.